src/mds/mdstypes.h

```diff
@@ -858,6 +866,65 @@ void inode_t<Allocator>::dump(Formatter *f) const
   f->dump_unsigned("last_scrub_version", last_scrub_version);
 }
 
+template<template<typename> class Allocator>
+void inode_t<Allocator>::client_ranges_cb(typename inode_t<Allocator>::client_range_map& c, JSONObj *obj){
+
+  int64_t client;
+  JSONDecoder::decode_json("client", client, obj, true);
+  client_writeable_range_t client_range_tmp;
+  JSONDecoder::decode_json("byte range", client_range_tmp.range, obj, true);
+  JSONDecoder::decode_json("follows", client_range_tmp.follows.val, obj, true);
+  c[client] = client_range_tmp;
+}
+
+template<template<typename> class Allocator>
+void inode_t<Allocator>::old_pools_cb(compact_set<int64_t, std::less<int64_t>, Allocator<int64_t> >& c, JSONObj *obj){
+
+  int64_t tmp;
+  decode_json_obj(tmp, obj);
+  c.insert(tmp);
+}
+
+template<template<typename> class Allocator>
+void inode_t<Allocator>::decode_json(JSONObj *obj)
+{
+
+  JSONDecoder::decode_json("ino", ino.val, obj, true);
+  JSONDecoder::decode_json("rdev", rdev, obj, true);
+  //JSONDecoder::decode_json("ctime", ctime, obj, true);
+  //JSONDecoder::decode_json("btime", btime, obj, true);
```

Comment:
why comment out these

Reply:
??

Reply:
@simon-rock

Reply:
@batrick I find a issue that the time parsed from the json file is several hours behind the original time. I guess this is due to the local time zone, but I haven't solved it yet.
I think there may be some potential requirements for using this function.

Reply:
These should be UNIX epochs so it's from UTC time. Timezones are ignored.

Reply:
you means this is a bug of JSONDecoder?If so. I will fix it later. 
Now there are some places to call this method